### ISSUE TITLE
Plug'n'Play does not work with Node 11

### ISSUE THREAD

**Issue by Turbo87** [yarnpkg/yarn#7092]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

When using `yarn run` on Node 11 it does not activate PnP mode automatically when `"pnp": true` is specified in the `package.json` file.

**If the current behavior is a bug, please provide the steps to reproduce.**

- create a new folder
- `yarn init`
- `yarn add chalk` (just an example)
- create an `index.js` file:

```js
let chalk = require('chalk');

console.log(chalk.red('hello'));
```

- add a script to `package.json` so that `yarn run foo` runs the `index.js` file
- `yarn run foo` (should run fine and print the string in red)
- `yarn --pnp` (enable PnP mode)
- `yarn run foo` (fails with `Error: Cannot find module 'chalk'`)

when you switch over from Node 11 to Node 10 everything works fine

**What is the expected behavior?**

`yarn run` should enable PnP mode automatically on Node 11 too

**Please mention your node.js, yarn and operating system version.**

OS: macOS 10.13.6
node: 10.15.0 and 11.10.0
yarn: 1.13.0

**Comment by rwjblue** [comment]
@stefanpenner, @davecombs, and myself ran across this issue (discussion over in https://github.com/stefanpenner/resolve-package-path/pull/7). The symptoms that we witness were that in Node 11 any scripts had to manually add `-r ./.pnp.js` to ensure the `.pnp.js` was required when ran via `yarn test` however the same setup worked without manually requiring `.pnp.js` when ran under Node 10.

**Comment by rwjblue** [comment]
I spent some time digging into this and found the underlying issue in Node 11's parsing of `NODE_OPTIONS`. I reported that here: https://github.com/nodejs/node/issues/26521.

However, I _also_ went to submit a patch working around the issue but I believe it is already fixed (by https://github.com/yarnpkg/yarn/pull/6942 and https://github.com/yarnpkg/yarn/pull/6951) on master. Unfortunately, https://github.com/yarnpkg/yarn/pull/6951 is not included in 1.14.0 (but https://github.com/yarnpkg/yarn/pull/6942 is) so on 1.14.0 _some_ things "work" with Node 11 (e.g. `yarn node some-file.js` or `yarn mocha tests/**.js`) but lifecycle scripts do not.

**Comment by arcanis** [comment]
Thanks for the investigation, @rwjblue! That was a good one 😅

I've started the release process for the 1.15 (and will put the 1.14 in stable once it's done), it was time to do it anyway 🙂 the patch should be made available in a few hours.

**Comment by Turbo87** [comment]
closing as this got fixed in Node v11.12.0
